engine/collection/synchronization/engine.go

```diff
@@ -307,7 +307,12 @@ func (e *Engine) onBlockResponse(originID flow.Identifier, res *messages.Cluster
 			OriginID: originID,
 			Block:    block,
 		}
-		e.comp.SubmitLocal(synced)
+		// forward the block to the compliance engine for validation and processing
+		// we use the network.MessageProcessor interface here because the block is un-validated
+		err := e.comp.Process(channels.SyncCluster(block.Header.ChainID), originID, synced)
```

Comment:
I think we need to pass me.NodeID() here instead of originID, original sender is already part of `SyncedClusterBlock`

Reply:
Yeah, I feel like we're mixing practices a bit here. 
* We decided to use `Process` API, because the messages have not been validated (should be treated as originating from another node)
* But passing `me.NodeID()` as `originID` indicates the message originates from this node (trusted)

The MessageProcessor interface says
https://github.com/onflow/flow-go/blob/6a39c8c64184231a05dd5c756db91307f0361e78/network/engine.go#L43-L47

Overall I think the API doesn't fit well with this usage pattern, where an untrusted message is passing through an internal component, without being validated first. Though setting `originID` to the local node ID for an untrusted message seems like the worse of two not-great options here.

Reply:
I am not sure here. We need to define clear rules how should we treat such cases. 
For instance sync engine doesn't do any checks so technically it's an untrusted input on other side it didn't come from network layer but from the sync engine. 
I would treat this case as general one as if message comes from unknown origin and only use trusted callbacks when we actually need different processing logic.

Reply:
OK, for now I will do that and pass `me.NodeID` with a comment.